### PR TITLE
vim-youcompleteme: fix python interpreter

### PR DESCRIPTION
**Pull request by LnL7**
###### Motivation for this change

The variables where renamed, perhaps it would be better to use a patch here instead.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -8606,6 +8606,7 @@ in
     inherit (darwin.apple_sdk.frameworks) Security;
   };
   rustracerd = callPackage ../development/tools/rust/racerd {
+    inherit (rustPackages_1_38_0) rustPlatform;
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix]: Is this a related change?

Comment (LnL7) [reply pkgs/top-level/all-packages.nix]: racerd is one of the dependencies used for rust conpletion, which broke with the recent update

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
ycmd seems to work

```
[4 built, 16 copied (2364.4 MiB), 512.6 MiB DL]
https://github.com/NixOS/nixpkgs/pull/73081
3 package were build:
rustracerd vimPlugins.youcompleteme ycmd
```
